internal/filters/tag_test.go

```diff
@@ -0,0 +1,116 @@
+package filters
+
+import (
+	"testing"
+
+	"github.com/kubewarden/sbomscanner/api/v1alpha1"
+	"github.com/kubewarden/sbomscanner/internal/cel"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_filterByTag(t *testing.T) {
+	te, err := cel.NewTagEvaluator()
+	imageTag := "1.27.1"
+	require.NoError(t, err)
+
+	tests := []struct {
+		name            string
+		tagEvaluator    *cel.TagEvaluator
+		matchConditions []v1alpha1.MatchCondition
+		tag             string
+		want            bool
+		wantErr         bool
+	}{
+		{
+			name:         "matches single condition",
+			tagEvaluator: te,
+			matchConditions: []v1alpha1.MatchCondition{
+				{
+					Name:       "no images with -dev tags",
+					Expression: "!tag.matches('$-dev')",
+				},
+			},
+			tag:     imageTag,
```

Comment:
nitpick: consider using the literal string value instead of the variable to make the test more explicit. That way readers don't need to jump to the variable definition to see what's being tested.